test/Chainweb/Test/CutDB.hs

```diff
@@ -486,3 +497,31 @@ fakePact = WebPactExecutionService $ PactExecutionService
     getFakeOutput (Transaction txBytes) = TransactionOutput txBytes
     coinbase = noCoinbaseOutput
     fakeMiner = MinerData "fakeMiner"
+
+tests :: RocksDb -> TestTree
+tests rdb = testGroup "CutDB"
+    [ testCutPruning rdb version
+    ]
+    where
+    version = Test petersonChainGraph
+
+testCutPruning :: RocksDb -> ChainwebVersion -> TestTree
+testCutPruning rdb v = testCase "cut pruning" $ do
+    let pruningThreshold = 100
+    -- initialize cut DB and mine enough to trigger pruning
+    let alterPruningSettings = 
+            set cutDbParamsAvgBlockHeightPruningDepth 50 .
+            set cutDbParamsAvgBlockHeightPruningThreshold pruningThreshold .
+            set cutDbParamsAvgBlockHeightWritingGap 10
+        minedBlockHeight = 201
+    (cutHashesStore, ()) <- withTestCutDbWithoutPact rdb v alterPruningSettings (int $ avgCutHeightAt v minedBlockHeight) (\_ _ -> return ()) (const $ return ())
```

Comment:
Why not move the assertions into the scope of the `withTestCutDbWithoutPact` bracket? That would also make above changes to the return value of `withTestCutDb` redundant.

Reply:
Done.